src/libsync/discoveryphase.cpp

```diff
@@ -100,156 +87,90 @@ bool DiscoveryJob::checkSelectiveSyncNewFolder(const QString &path, RemotePermis
+            startJob(nextJob);
         } else {
-            updateJob->_lastUpdateProgressCallbackCall.start();
-        }
-
-        QByteArray pPath(dirUrl);
-        int indx = pPath.lastIndexOf('/');
-        if (indx > -1) {
-            const QString path = QUrl::fromPercentEncoding(pPath.mid(indx + 1));
-            emit updateJob->folderDiscovered(local, path);
+            emit finished();
         }
-    }
+    });
+    _currentRootJob = job;
+    job->start();
 }
 
-// Only use for error cases! It will always set an error errno
-int get_errno_from_http_errcode(int err, const QString &reason)
+void DiscoveryPhase::scheduleMoreJobs()
 {
-    int new_errno = EIO;
-
-    switch (err) {
-    case 401: /* Unauthorized */
-    case 402: /* Payment Required */
-    case 407: /* Proxy Authentication Required */
-    case 405:
-        new_errno = EPERM;
-        break;
-    case 301: /* Moved Permanently */
-    case 303: /* See Other */
-    case 404: /* Not Found */
-    case 410: /* Gone */
-        new_errno = ENOENT;
-        break;
-    case 408: /* Request Timeout */
-    case 504: /* Gateway Timeout */
-        new_errno = EAGAIN;
-        break;
-    case 423: /* Locked */
-        new_errno = EACCES;
-        break;
-    case 403: /* Forbidden */
-        new_errno = ERRNO_FORBIDDEN;
-        break;
-    case 400: /* Bad Request */
-    case 409: /* Conflict */
-    case 411: /* Length Required */
-    case 412: /* Precondition Failed */
-    case 414: /* Request-URI Too Long */
-    case 415: /* Unsupported Media Type */
-    case 424: /* Failed Dependency */
-    case 501: /* Not Implemented */
-        new_errno = EINVAL;
-        break;
-    case 507: /* Insufficient Storage */
-        new_errno = ENOSPC;
-        break;
-    case 206: /* Partial Content */
-    case 300: /* Multiple Choices */
-    case 302: /* Found */
-    case 305: /* Use Proxy */
-    case 306: /* (Unused) */
-    case 307: /* Temporary Redirect */
-    case 406: /* Not Acceptable */
-    case 416: /* Requested Range Not Satisfiable */
-    case 417: /* Expectation Failed */
-    case 422: /* Unprocessable Entity */
-    case 500: /* Internal Server Error */
-    case 502: /* Bad Gateway */
-    case 505: /* HTTP Version Not Supported */
-        new_errno = EIO;
-        break;
-    case 503: /* Service Unavailable */
-        // https://github.com/owncloud/core/pull/26145/files
-        if (reason == "Storage not available" || reason == "Storage is temporarily not available") {
-            new_errno = ERRNO_STORAGE_UNAVAILABLE;
-        } else {
-            new_errno = ERRNO_SERVICE_UNAVAILABLE;
-        }
-        break;
-    case 413: /* Request Entity too Large */
-        new_errno = EFBIG;
-        break;
-    default:
-        new_errno = EIO;
+    auto limit = qMax(1, _syncOptions._parallelNetworkJobs);
+    if (_currentRootJob && _currentlyActiveJobs < limit) {
+        _currentRootJob->progress(limit - _currentlyActiveJobs);
```

Comment:
Hm. I think it's essential that it gets renamed since currently there's `progress()` and `process()` and that's quite confusing. One is for processing the directory itself, one is for processing the sub jobs that were spawned.

Maybe `processSubJobs()` then, with some added documentation? It seems dealing with the queued up jobs is the second phase of ProcessDirectoryJob. Maybe the two aspects could even be disentangled further?